File: Array&String/transpose_matrix.py
class Solution:
    """
    Task:
        Given a 2D integer array matrix, return the transpose of matrix.
        The transpose of a matrix is the matrix flipped over its main diagonal, switching the matrix's row and column indices.
    """
    def transpose(self, matrix: list[list[int]]) -> list[list[int]]:
        return [[matrix[i][j] for i in range(len(matrix))] for j in range(len(matrix[0]))]

File: Array-String/CPP/transpose_matrix.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        vector<vector<int>> ans;
        for (int j = 0; j < n; j++) {
            vector<int> cur;
            for (int i = 0; i < m; i++) {
                cur.push_back(matrix[i][j]);
            }
            ans.push_back(cur);
        }
        return ans;
    }
};